synfig-studio/plugins/lottie-exporter/common/Param.py

```diff
@@ -487,6 +487,37 @@ def recur_animate(self, anim_type):
                 self.expression = ret
                 return ret, self.expression_controllers
 
+            elif self.param[0].tag == "cos":
+                self.subparams["cos"].extract_subparams()
+                angle, eff_1 = self.subparams["cos"].subparams["angle"].recur_animate("region_angle")
+                amp, eff_2 = self.subparams["cos"].subparams["amp"].recur_animate("real")
+                self.expression_controllers.extend(eff_1)
+                self.expression_controllers.extend(eff_2)
+                
+                if self.dimension == 2:
+                    ret = "mul(Math.cos(degreesToRadians({angle})), [{amp}, {amp}])"
+                else:
+                    ret = "mul(Math.cos(degreesToRadians({angle})),{amp})"
+                ret = ret.format(angle=angle,amp=amp)
+
+                self.expression = ret
+                return ret, self.expression_controllers
+
+            elif self.param[0].tag == "fromint":
+                self.subparams["fromint"].extract_subparams()
+                if self.dimension == 2:
+                    link, eff_1 = self.subparams["fromint"].subparams["link"].recur_animate("points")
+                    self.expression_controllers.extend(eff_1)
+                    ret = "[{link},{link}]"
```

Comment:
Here simple circle implementation is giving different output than expected.
I am attaching the synfig file on which it fails:
[integer_circle.zip](https://github.com/synfig/synfig/files/4350165/integer_circle.zip)

Reply:
I suspect this is happening because you are using `animation type` as `real` and `points`, whereas we need something discrete.

Reply:
BTW, why have you used different `anim_type` in the `if-else` conditions?

Reply:
Thanks for the clarification. This is resolved now!